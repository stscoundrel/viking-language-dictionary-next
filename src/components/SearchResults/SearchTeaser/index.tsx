import { SearchResult } from 'lib/services/search'
import Link from 'next/link'
import styles from './SearchTeaser.module.scss'

interface SearchTeaserProps{
  data: SearchResult
}

export default function SearchTeaser({ data }: SearchTeaserProps) {
  const { slug, headword, foundIn } = data

  return (
    <div className={styles.result}>
      <Link key={`link${slug}`} href={`/word/${slug}`}>
        <a className={styles.link}>{headword.toLowerCase()}</a>
      </Link>
      <ul>
        { foundIn.map((searchResult, index) => (
          <li
            className={styles.foundIn}
            key={`${data.slug}-search-${index}`}
            dangerouslySetInnerHTML={ { __html: searchResult } }
          />
        )) }
      </ul>
    </div>
  )
}
