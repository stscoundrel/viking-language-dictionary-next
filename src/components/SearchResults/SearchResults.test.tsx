import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import SearchResults from './index'

const words = [
  {
    headword: 'þess',
    definition: 'm/n gen sg of sá/þat',
    isCommon: false,
    slug: 'thess',
    foundIn: ['In headword'],
  },
  {
    headword: 'þessi',
    definition: '(also sjá) <f þessi, n þetta, m acc sg þenna, m dat sg & all dat pl þessum~þeima, m/n gen sg þessa, m nom pl þessir, all gen pl þessa~þessar(r)a, f acc sg þessa, f dat sg þessi~þessar(r)i, f gen sg þessar~þessar(r)ar, n dat sg þessu~þvísa, n nom/acc pl þessi> dem pron this, these',
    isCommon: true,
    slug: 'thessi',
    foundIn: ['In headword'],
  },
  {
    headword: 'þetta',
    definition: 'n nom/acc sg of þessi',
    isCommon: false,
    slug: 'thetta',
    foundIn: ['In headword'],
  },
  {
    headword: 'þér',
    definition: '<acc/dat yðr, gen yðarr~yðvarr> pl pron you; þér frequent dat of þú; see ér',
    isCommon: false,
    slug: 'ther',
    foundIn: ['In headword'],
  },
  {
    headword: 'Þiðrandi',
    definition: 'm Thidrandi (personal name)',
    isCommon: false,
    slug: 'thidrandi',
    foundIn: ['In headword'],
  },
  {
    headword: 'þiggja',
    definition: '<þiggr, þá, þágu, þeginn> vb (V) to accept; receive; accept lodgings',
    isCommon: false,
    slug: 'thiggja',
    foundIn: ['In headword'],
  },
  {
    headword: 'þinglogi',
    definition: 'adj indecl engagement breaker',
    isCommon: false,
    slug: 'thinglogi',
    foundIn: ['In headword'],
  },
]

describe('SearchResults component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<SearchResults words={words} />)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<SearchResults words={words} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
