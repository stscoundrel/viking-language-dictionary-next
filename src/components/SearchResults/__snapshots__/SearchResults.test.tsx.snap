// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component Matches snapshot 1`] = `
Array [
  <p
    className="blue"
  >
    7
     results found
  </p>,
  <ul
    className="list"
  >
    <li>
      <div
        className="result"
      >
        <a
          className="link"
          href="/word/thess"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          þess
        </a>
        <ul>
          <li
            className="foundIn"
            dangerouslySetInnerHTML={
              Object {
                "__html": "In headword",
              }
            }
          />
        </ul>
      </div>
    </li>
    <li>
      <div
        className="result"
      >
        <a
          className="link"
          href="/word/thessi"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          þessi
        </a>
        <ul>
          <li
            className="foundIn"
            dangerouslySetInnerHTML={
              Object {
                "__html": "In headword",
              }
            }
          />
        </ul>
      </div>
    </li>
    <li>
      <div
        className="result"
      >
        <a
          className="link"
          href="/word/thetta"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          þetta
        </a>
        <ul>
          <li
            className="foundIn"
            dangerouslySetInnerHTML={
              Object {
                "__html": "In headword",
              }
            }
          />
        </ul>
      </div>
    </li>
    <li>
      <div
        className="result"
      >
        <a
          className="link"
          href="/word/ther"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          þér
        </a>
        <ul>
          <li
            className="foundIn"
            dangerouslySetInnerHTML={
              Object {
                "__html": "In headword",
              }
            }
          />
        </ul>
      </div>
    </li>
    <li>
      <div
        className="result"
      >
        <a
          className="link"
          href="/word/thidrandi"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          þiðrandi
        </a>
        <ul>
          <li
            className="foundIn"
            dangerouslySetInnerHTML={
              Object {
                "__html": "In headword",
              }
            }
          />
        </ul>
      </div>
    </li>
    <li>
      <div
        className="result"
      >
        <a
          className="link"
          href="/word/thiggja"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          þiggja
        </a>
        <ul>
          <li
            className="foundIn"
            dangerouslySetInnerHTML={
              Object {
                "__html": "In headword",
              }
            }
          />
        </ul>
      </div>
    </li>
    <li>
      <div
        className="result"
      >
        <a
          className="link"
          href="/word/thinglogi"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          þinglogi
        </a>
        <ul>
          <li
            className="foundIn"
            dangerouslySetInnerHTML={
              Object {
                "__html": "In headword",
              }
            }
          />
        </ul>
      </div>
    </li>
  </ul>,
]
`;
